bindingcollectionadapter/src/main/java/me/tatarka/bindingcollectionadapter2/itembindings/OnItemBindClass.java

```diff
@@ -42,16 +57,27 @@ public int itemTypeCount() {
         return itemBindingMap.size();
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void onItemBind(ItemBinding itemBinding, int position, T item) {
-        for (int i = 0; i < itemBindingMap.size(); i++) {
-            Class<? extends T> key = itemBindingMap.keyAt(i);
+        for (int i = 0; i < itemBindingIndexes.size(); i++) {
+            Class<? extends T> key = itemBindingIndexes.get(i);
             if (key.isInstance(item)) {
-                int[] values = itemBindingMap.valueAt(i);
-                itemBinding.set(values[0], values[1]);
+                onTypedItemBind(itemBinding, position, item, itemBindingMap.get(key));
                 return;
             }
         }
         throw new IllegalArgumentException("Missing class for item " + item);
     }
+
+    @SuppressWarnings("unchecked")
+    private <E extends T> void onTypedItemBind(ItemBinding itemBinding, int position, E item, Object itemBind) {
+        if (itemBind instanceof OnItemBind) {
```

Comment:
Wouldn't this logic be simpler if you always had an `onItemBind`? You can construct one from the `variableId` and `layoutRes`.